### ISSUE TITLE
S3 Integration

### ISSUE THREAD

**Issue by varblob** [benefitcloud/ember-uploader#94]
I'm attempting to get S3 working and am struggling a bit.  It looks like perhaps things have changed.  

Here is what I have for signing at the moment still getting 401 forbidden but that could be something else.

    bucket_name = "bucket_name"
    bucket = Aws::S3::Resource.new(region: 'region').bucket(bucket_name)
    @s3_direct_post = bucket.presigned_post(
      key: "path to where you want to put it/${filename}", #${filename} is amazon magic
      success_action_status: "201",
      acl: "public-read",
      expires: 1.hours.from_now
    )
    # bucket name appears to be needed by the client in composing the url
    render json: @s3_direct_post.fields.merge(bucket: bucket_name), status: :ok

**Comment by digitaltoad** [comment]
You are getting a 401 from the sign?

**Comment by varblob** [comment]
Sorry 403 from attempting to upload.

**Comment by digitaltoad** [comment]
What is the body of the error?

**Comment by varblob** [comment]
Not very informative 

    <Error>
    <Code>AccessDenied</Code>
    <Message>Access Denied</Message>
    <RequestId>......</RequestId>
    <HostId>........</HostId>
    </Error>

**Comment by digitaltoad** [comment]
Seems like it may be related to something like this. https://forums.aws.amazon.com/thread.jspa?threadID=56531

**Comment by varblob** [comment]
Thanks for your help, I think I need to fix my policy file for the user.  I'll let you know if it all works out.

**Comment by digitaltoad** [comment]
That being said. I don't think there is anything wrong with how you are currently generating the policy rather there is something wrong with the aws user's rights.

**Comment by varblob** [comment]
yup user needed to have his rights updated, success!  I'm leaving this issue open since the original bit maybe of use to others.  If I have time next week I'll try and make a pull request for the documentation.

**Comment by digitaltoad** [comment]
I may get to the documentation before that. I'm currently in the process of updating the outdated parts of the library including making the base library ember-cli compatible so the wrapper can be deprecated. Updating the docs around signing was on the list.

**Comment by varblob** [comment]
most awesome!  thanks again for your help and for the lib.  I renamed this issue to reference documentation.
